Avoid dependency on RNR-initialisation

meriac opened the pull request:
In case the RNR register is initialised to a region outside of the supported region count the core behaviour is undefined by the architecture. To avoid that we specifically enforce region number zero.

@AlessandroA @Patater

AlessandroA commented on core/system/src/mpu/vmpu_armv7m.c:
```diff
@@ -68,8 +68,7 @@
 /* MPU helper macros */
-#define MPU_RBAR(region,addr)   (((uint32_t)(region))|MPU_RBAR_VALID_Msk|addr)
-#define MPU_RBAR_RNR(addr)     (addr)
+#define MPU_RBAR_RNR(region)      (MPU_RBAR_VALID_Msk | (((region)<<MPU_RBAR_REGION_Pos) & MPU_RBAR_REGION_Msk))
```
Could you add spaces around `<<`?

AlessandroA commented on core/system/src/mpu/vmpu_armv7m.c:
```diff
@@ -629,10 +628,11 @@ void vmpu_arch_init(void)
     MPU->CTRL = 0;
-    /* check MPU region aligment */
-    MPU->RBAR = MPU_RBAR_ADDR_Msk;
+    /* check MPU region aligment using region 0 */
```
`Check MPU region alignment using region 0.` (we now use always capitalized, punctuated sentences) (also, mind `aligNment`)

Patater commented:
If there is a GitHub issue this fixes, mention the issue in the commit message.

Patater commented on core/system/src/mpu/vmpu_armv7m.c:
```diff
@@ -629,10 +628,11 @@ void vmpu_arch_init(void)
-    /* check MPU region aligment */
-    MPU->RBAR = MPU_RBAR_ADDR_Msk;
+    /* check MPU region aligment using region 0 */
+    MPU->RNR = 0;
```
Do we need to program the RNR to something else later, or is leaving it at 0 okay?

meriac replied:
@Patater leaving it at zero is OK, I wourd rather not want to depend on other code re-initializing this register.

Patater commented:
LGTM

Patater commented on core/system/src/mpu/vmpu_armv7m.c:
```diff
@@ -639,12 +639,14 @@ void vmpu_arch_init(void)
     /* sanity checks */
-    assert(ARMv7M_MPU_REGIONS == ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF));
-    assert(ARMv7M_MPU_ALIGNMENT_BITS == vmpu_bits(g_vmpu_aligment_mask));
+    if(ARMv7M_MPU_REGIONS != ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF))
```
Please do these assert changes in a separate commit.

meriac commented:
@AlessandroA @Patater  simplified patch
- removed new MPU_RBAR macro, so `<<`-whitespace-fix is not necessary any more
- split into separate functional patches

Patater commented:
LGTM

AlessandroA commented on core/system/src/mpu/vmpu_armv7m.c:
```diff
@@ -640,13 +640,14 @@ void vmpu_arch_init(void)
-    assert(ARMv7M_MPU_REGIONS == ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF));
-    assert(ARMv7M_MPU_ALIGNMENT_BITS == vmpu_bits(g_vmpu_aligment_mask));
-
+    if(ARMv7M_MPU_REGIONS != ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF))
```
Could you add a space after the `if` and the [1TBS braces](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)?

```C
if (condition) {
    code;
}
```

(also applies to line 649)

AlessandroA commented:
Added [a comment](https://github.com/ARMmbed/uvisor/pull/194/commits/d255b615b9736061710e4f75860d8fb7617a728c#r56215974). Other than that, LGTM.

meriac commented:
@AlessandroA added braces to both HALT_ERROR, please merge if you're happy with it.

AlessandroA commented:
Thanks! LGTM
